platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -409,12 +409,24 @@ class ConfigurationCacheProblems(
     }
 
     private
-    fun areDegradationReasonsPresent(): Boolean = degradationController.degradationReasons.isNotEmpty()
+    fun areDegradationReasonsPresent(): Boolean = degradationController.hasDegradationReasons
+
+    private
+    fun areTaskDegradationReasonsPresent(): Boolean = degradationController.hasTaskDegradationReasons
 
     private
     fun degradationSummary(): String {
-        val degradingTasks = degradationController.degradationReasons.keys
-        return " because incompatible ${if (degradingTasks.size > 1) "tasks were" else "task was"} found."
+        val degradingTasks = degradationController.taskDegradationReasons.size
+        val degradingFeatures = degradationController.featureDegradationReasons
+        val featureList = degradingFeatures.map { it.incompatibleFeatureName }.joinToString().let { "($it)" }
+        return " because incompatible " +
+            when {
+                degradingTasks == 1 && degradingFeatures.isEmpty() -> "task was"
+                degradingTasks > 1 && degradingFeatures.isEmpty() -> "tasks were"
+                degradingTasks == 0 && degradingFeatures.size ==  1 -> "feature ${featureList} was"
```

Comment:
💅  I think that wording like "feature was found" doesn't fit here well, whereas it's good for the tasks. "Feature usage" can be found, or something like this, but not the feature by itself